File: leetcode/ag_01.c
#include <stdio.h>

int * twoSum(int* nums, int numsSize, int target) {
//    int ans[2];
//    for(int i = 0; i<numsSize; i++){
//        for (int j = i+1 ; j<numsSize ; j++){
//            if(nums[i] + nums[j] == target){
//                ans[0] = i;
//                ans[1] = j;
//                break;
//            }
//        }
//    }
//    return ans;

    int ans[2];
    for(int i = 0; i<numsSize; i++){
        for (int j = i+1 ; j<numsSize ; j++){
            if(nums[i] + nums[j] == target){
                ans[0] = i;
                ans[1] = j;
                break;
            }
        }
    }
    return ans;

}

int main()
{
    int a[4]={0,4,3,0};
    int * ans = twoSum(a,4,0);
    printf("%d , %d\n",ans[0],ans[1]);
    return 0;
}

File: leetcode/ag_01.py
#!/bin/python2.7

class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        ans = None
        for k,v in enumerate(nums):
            for k1,v1 in enumerate(nums[k+1:]):
                if v + v1 == target:
                    ans = k,k1+k+1
        return ans


def run():
     test = [13,  11, 7, 15 ,2]
     target = 9
     s = Solution()
     print s.twoSum(test , target)
